src/vs/workbench/parts/welcome/page/electron-browser/welcomePage.ts

```diff
@@ -61,8 +62,21 @@ export class WelcomePageContribution implements IWorkbenchContribution {
 			backupFileService.hasBackups().then(hasBackups => {
 				const activeEditor = editorService.activeEditor;
 				if (!activeEditor && !hasBackups) {
-					return instantiationService.createInstance(WelcomePage)
-						.openEditor();
+					const openWithReadme = configurationService.getValue(configurationKey) === 'openWithReadme';
+					if (openWithReadme) {
+						const onlyWorkSpaceFolder = contextService.getWorkbenchState() === WorkbenchState.FOLDER
```

Comment:
It makes sense for this to work when there are multiple folders too. Maybe just check all folders and open the first README that is found.